src/Login.tsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import { useState } from 'react';
+import {
+  Text,
+  Button,
+  TextInput,
+  PasswordInput,
+} from '@mantine/core';
+import { useForm, Controller } from 'react-hook-form';
+
+interface MyFormValues {
+  password: string;
+  email: string;
+  confirmPassword?: string;
+}
+
+export default function Login() {
+  const [isRegistered, setIsRegistered] = useState(true);
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+    getValues,
+  } = useForm<MyFormValues>({
+    defaultValues: {
+      email: '',
+      password: '',
+      confirmPassword: '',
+    },
+  });
+
+  const handleChangeIsRegistered = (): void => {
+    setIsRegistered(!isRegistered)
+  }
+
+  const onSubmit = (data: MyFormValues) => {
+    console.log('email: ', data.email);
+    console.log('password: ', data.password);
+  }
+
+  return (
+    <div>
+      {!!isRegistered ? 
+      <>
+        <Text>Log In</Text>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <Controller
```

Comment:
shouldn't need the controller, you should be able to subscribe the field to react-hook-forms directly

Reply:
haha, kk, this was a pretty big refactor, but I do understand how react hook forms works a bit better now I think. Think I got it now